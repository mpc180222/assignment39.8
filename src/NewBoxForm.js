import React, { useState } from 'react';
import './App.css';
import BoxList from './BoxList'


const NewBoxForm = ({addBox}) => {

    const INITIAL_STATE = {
        width: '',
        height: '',
        color: ''

    }

    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleSubmit = (e) => {
        e.preventDefault();
        // addBox(formData.width, formData.height, formData.color);
        addBox({widthData: formData.width, heightData: formData.height, colorData: formData.color});
        setFormData(INITIAL_STATE);
    }

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(formData => ({
            ...formData, [name]: value

        }))

    }

    return(

        <form onSubmit = {handleSubmit}>
            <input data-testid="new-box-width-input" name = 'width' id = 'width' type = 'text' placeholder='width' value = {formData.width} onChange = {handleChange}></input>
            <input data-testid="new-box-height-input" name = 'height' id = 'height' type = 'text' placeholder='height' value = {formData.height} onChange = {handleChange}></input>
            <input data-testid="new-box-color-input" name = 'color' id = 'color' type = 'text' placeholder='color' value = {formData.color} onChange = {handleChange}></input>
            <button data-testid="new-box-submit-btn">Add box</button>
        </form>

    )




}

export default NewBoxForm;