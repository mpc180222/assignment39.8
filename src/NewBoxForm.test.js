import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

it('renders without crashing', () => {
  render(<NewBoxForm/>);
  
});

it("should match snapshot", () => {
  const { asFragment } = render(<NewBoxForm />)
  expect(asFragment()).toMatchSnapshot();
})