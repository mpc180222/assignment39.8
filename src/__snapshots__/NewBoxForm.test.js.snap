// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      data-testid="new-box-width-input"
      id="width"
      name="width"
      placeholder="width"
      type="text"
      value=""
    />
    <input
      data-testid="new-box-height-input"
      id="height"
      name="height"
      placeholder="height"
      type="text"
      value=""
    />
    <input
      data-testid="new-box-color-input"
      id="color"
      name="color"
      placeholder="color"
      type="text"
      value=""
    />
    <button
      data-testid="new-box-submit-btn"
    >
      Add box
    </button>
  </form>
</DocumentFragment>
`;
